File: src/test/java/it/marketplace/microservices/common/resource/ProductOrderResourceTest.java
package it.marketplace.microservices.common.resource;

import org.junit.jupiter.api.Test;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import static org.junit.jupiter.api.Assertions.assertEquals;

class ProductOrderResourceTest {
    @Test
    void shouldCreateProductOrderResourceWithAllFields() {
        // Arrange
        LocalDateTime creationDate = LocalDateTime.now();
        LocalDateTime tmsUpdate = LocalDateTime.now();
        // Act
        ProductOrderResource resource = new ProductOrderResource("ORD123", "PROD1", BigDecimal.ONE, 10.0, 10.0, creationDate, tmsUpdate);
        // Assert
        assertEquals("ORD123", resource.getOrderCode());
        assertEquals("PROD1", resource.getProductCode());
        assertEquals(BigDecimal.ONE, resource.getQuantity());
        assertEquals(10.0, resource.getUnitPrice());
        assertEquals(10.0, resource.getTotal());
        assertEquals(creationDate, resource.getCreationDate());
        assertEquals(tmsUpdate, resource.getTmsUpdate());
    }

    @Test
    void shouldSetAndGetTmsUpdate() {
        // Arrange
        ProductOrderResource resource = new ProductOrderResource();
        LocalDateTime tmsUpdate = LocalDateTime.now();
        // Act
        resource.setTmsUpdate(tmsUpdate);
        // Assert
        assertEquals(tmsUpdate, resource.getTmsUpdate());
    }
}

File: src/test/java/it/marketplace/microservices/common/resource/ProductOrderResourceTest.kt
package it.marketplace.microservices.common.resource

import org.junit.jupiter.api.Test
import java.math.BigDecimal
import java.time.LocalDateTime
import kotlin.test.assertEquals

class ProductOrderResourceTest {
    @Test
    fun shouldCreateProductOrderResourceWithAllFields() {
        // Arrange
        val creationDate = LocalDateTime.now()
        val tmsUpdate = LocalDateTime.now()
        // Act
        val resource = ProductOrderResource(1L, "ORD123", "PROD1", BigDecimal.ONE, 10.0, 10.0, creationDate, tmsUpdate)
        // Assert
        assertEquals("ORD123", resource.orderCode)
        assertEquals("PROD1", resource.productCode)
        assertEquals(BigDecimal.ONE, resource.quantity)
        assertEquals(10.0, resource.unitPrice)
        assertEquals(10.0, resource.total)
        assertEquals(creationDate, resource.creationDate)
        assertEquals(tmsUpdate, resource.tmsUpdate)
    }

    @Test
    fun shouldSetAndGetTmsUpdate() {
        // Arrange
        val resource = ProductOrderResource(
            id = 1L,
            orderCode = "ORD123",
            productCode = "PROD1",
            quantity = BigDecimal.ONE,
            unitPrice = 10.0,
            total = 10.0,
            creationDate = LocalDateTime.now(),
            tmsUpdate = LocalDateTime.now()
        )
        val tmsUpdate = LocalDateTime.now()
        // Act
        resource.tmsUpdate = tmsUpdate
        // Assert
        assertEquals(tmsUpdate, resource.tmsUpdate)
    }
}

